test/Microsoft.AspNetCore.Sockets.Client.Tests/HubConnectionTests.cs

```diff
@@ -167,5 +165,62 @@ public void CannotCreateHubConnectionWithNullUrl()
                 Assert.Null(await closedEventTcs.Task.OrTimeout());
             }
         }
+
+        [Fact]
+        public async Task CannotCallInvokeOnClosedHubConnection()
+        {
+            var mockConnection = new Mock<IConnection>();
+            mockConnection
+                .Setup(m => m.DisposeAsync())
+                .Callback(() => mockConnection.Raise(c => c.Closed += null, (Exception)null))
+                .Returns(Task.FromResult<object>(null));
+
+            var hubConnection = new HubConnection(mockConnection.Object, Mock.Of<IInvocationAdapter>(), new LoggerFactory());
+
+            await hubConnection.StartAsync(Mock.Of<ITransport>());
+            await hubConnection.DisposeAsync();
+            var exception = await Assert.ThrowsAsync<InvalidOperationException>(
+                async () => await hubConnection.Invoke("test", typeof(int)));
+
+            Assert.Equal("Connection has been terminated.", exception.Message);
+        }
+
+        [Fact]
+        public async Task PendingInvocationsAreCancelledWhenConnectionClosesCleanly()
+        {
+            var mockConnection = new Mock<IConnection>();
+            mockConnection
+                .Setup(m => m.DisposeAsync())
+                .Callback(() => mockConnection.Raise(c => c.Closed += null, (Exception)null))
+                .Returns(Task.FromResult<object>(null));
+
+            var hubConnection = new HubConnection(mockConnection.Object, Mock.Of<IInvocationAdapter>(), new LoggerFactory());
+
+            await hubConnection.StartAsync(Mock.Of<ITransport>());
+            var invokeTask = hubConnection.Invoke("testMethod", typeof(int));
```

Comment:
Actually, no. There is no server so no one can complete this invocation/invokeTask.